library/common/jni/jni_interface.cc

```diff
@@ -259,10 +259,18 @@ static envoy_filter_data_status jvm_http_filter_on_request_data(envoy_data data,
   JNIEnv* env = get_env();
   jobjectArray result = static_cast<jobjectArray>(
       jvm_on_data("onRequestData", data, end_stream, const_cast<void*>(context)));
+  jsize size = env->GetArrayLength(result);
 
   jobject status = env->GetObjectArrayElement(result, 0);
   jobject j_data = static_cast<jobjectArray>(env->GetObjectArrayElement(result, 1));
 
+  envoy_headers* pending_headers = nullptr;
+  if (size == 3) {
```

Comment:
sorry, I am a bit decontextualized here. Why 3? Can we have a constant?

Reply:
Yeah, this isn't a great check. I'd like to move to fully-bridged types for filter statuses. But this ensures we don't attempt out of bounds access on the array.

Reply:
Sure the comment helps. Can we state why 3 here and 4 below? Maybe make them constants?

Reply:
Sure, I'll do that.

Reply:
Updated with a different and clearer check.